src/main/java/ch/jalu/configme/properties/types/BeanPropertyType.java

```diff
@@ -0,0 +1,54 @@
+package ch.jalu.configme.properties.types;
+
+import ch.jalu.configme.beanmapper.DefaultMapper;
+import ch.jalu.configme.beanmapper.Mapper;
+import ch.jalu.configme.resource.PropertyReader;
+import ch.jalu.configme.utils.TypeInformation;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@SuppressWarnings("unchecked")
+public class BeanPropertyType<B> implements PropertyType<B> {
+
+    private final TypeInformation beanType;
+    private final Mapper mapper;
+
+    // Caching by type and mapper
+    private static final Map<Class<?>, Map<Mapper, BeanPropertyType<?>>> CACHE = new HashMap<>();
+
+    private BeanPropertyType(TypeInformation beanType, Mapper mapper) {
+        this.beanType = beanType;
+        this.mapper = mapper;
+    }
+
+    @Override
+    public B get(PropertyReader reader, String path) {
+        return (B) mapper.convertToBean(reader, path, this.beanType);
+    }
+
+    @Override
+    public B convert(Object object) {
+        throw new UnsupportedOperationException();
```

Comment:
We should probably change the mapper to expect an Object as argument and not know about the `reader`. This is a change I can still do in 1.0 so we avoid a breaking change in 1.1.

Reply:
Please do it yourself, because I am not read packet `beanmapper`

Reply:
Yes, I'll do it today~ :sparkles:

Reply:
Done on master with b22e6b0